src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -1059,6 +1088,23 @@ impl BatchPlanFragmenter {
             .transpose()
     }
 
+    fn collect_stage_file_scan(node: PlanRef) -> SchedulerResult<Option<FileScanInfo>> {
+        if node.node_type() == PlanNodeType::BatchExchange {
+            // Do not visit next stage.
+            return Ok(None);
+        }
+
+        if let Some(_batch_file_scan) = node.as_batch_file_scan() {
+            // Currently the file scan only support one file, so we just need a empty struct.
```

Comment:
If we support scanning multiple files in the future, will there be multiple `FileScanNode`s or a single `FileScanNode` with an array of `file_location`?

Reply:
It should be a single FileScanNode with an array of file_location.

Reply:
I see. IIRC, a field can be changed into `repeated` with preserving backward compatibility. If so, no problem.

Reply:
We won't break any compatibility even if we change any fields in the future, because it is a batch plan node proto. So don't worry.